### PR TITLE
implemented basic error handling

### PR DESCRIPTION
**Pull request by miyamoto-san**

### CODE REVIEW

File: server/panels/router.js [hunk 1]
```diff
@@ -6,51 +6,51 @@ const path = require("path");
...
   }
 });
 
-function corsAllow(res, req) {
-  res.header("Access-Control-Allow-Origin", req.headers["origin"] || "*");
-  res.header("Access-Control-Allow-Credentials", "true");
-  res.header("Access-Control-Allow-Methods", "POST, GET, OPTIONS");
-  //res.header("Access-Control-Allow-Headers", "X-Requested-With");
-  res.header("Vary", "Origin");
-}
```
Comment (kaj) [inline server/panels/router.js:56]: This function, and the calls to it, are still needed.  Or should those headers be set in some other way?

File: server/server.js [hunk 2]
```diff
@@ -47,18 +47,33 @@ app.get("/kpm/logout", (req, res) => {
...
-      footer,
-      megaMenu,
-      search,
-    })
-  );
+app.get("/kpm", async (req, res) => {
+  try {
+    const footer = await fetchBlock("footer");
+    const megaMenu = await fetchBlock("megaMenu");
+    const search = await fetchBlock("search");
```
Comment (kaj) [inline server/server.js:55]: This also seems like a rebase error, as this reverts the caching PR by @exacs that was previously merged.

File: server/login-router.js [hunk 3]
```diff
@@ -50,17 +50,22 @@ async function authenticateUser(req) {
...
+    const loginUrl = new URL(`${process.env.SSO_ROOT_URL}/login`);
+    loginUrl.searchParams.set("service", serviceUrl);
 
-  log.info(`Redirect user to ${loginUrl}`);
-  res.redirect(loginUrl);
+    log.info(`Redirect user to ${loginUrl}`);
+    res.redirect(loginUrl);
+  } catch (err) {
+    log.error(err);
+    return res.status(400).send("");
```
Comment (kaj) [inline server/login-router.js:67]: How is any unexpected exception caught in the handler functions a request error, while similarly unexpected exceptions caught in the catchAll are server errors?

**Review by kaj — CHANGES_REQUESTED** [review]
Two things I think are rebase mistakes and one question about the actual error handling:

File: server/login-router.js (hunk 3 above)
Comment (exacs) [reply server/login-router.js:67]: Agree here. I think unexpected exceptions should not be handled in routes but in the `catchAll`

File: server/panels/router.js [hunk 4]
```diff
@@ -1,20 +1,25 @@
 const log = require("skog");
 const { Router } = require("express");
-const handlebars = require("handlebars");
-const fs = require("fs");
-const path = require("path");
 const { compileTemplate } = require("../utils");
 
 const panelsRouter = Router();
-module.exports = panelsRouter;
```
Comment (exacs) [inline server/panels/router.js:9]: This was move down. I don't have any opinions on that :)

**Review by kaj — APPROVED** [review]
Much smaller fixes now.  :-)  Seems ok to me from reading the code.

**Pull request merged**